src/assign.c

```diff
@@ -1004,8 +1004,8 @@ const char *memrecycle(const SEXP target, const SEXP where, const int start, con
     case REALSXP:
       if (sourceIsI64)
                     BODY(int64_t, REAL, int, (val==NA_INTEGER64||val>INT_MAX||val<=NA_INTEGER) ? NA_INTEGER : (int)val,  td[i]=cval)
```

Comment:
```suggestion
                    BODY(int64_t, REAL, int, (val==NA_INTEGER64||val>INT_MAX||val<=INT_MIN) ? NA_INTEGER : (int)val,  td[i]=cval)
```

same as above (although R will probably never change that INT_MIN = NA_INTEGER)